File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.go
func subsetXORSum(nums []int) int {
    var sum int;

    var dfs func(index, curXor int);

    dfs = func(index, curXor int) {
        if index == len(nums) {
            sum += curXor;

            return;
        }

        dfs(index + 1, curXor ^ nums[index]);
        dfs(index + 1, curXor);
    }

    dfs(0, 0);

    return sum;
}

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.java
class Solution {
    int sum = 0;

    public int subsetXORSum(int[] nums) {
        dfs(nums, 0, 0);

        return sum;
    }

    private void dfs(int[] nums, int i, int curXor) {
        if (i == nums.length) {
            sum += curXor;

            return;
        }

        dfs(nums, i + 1, curXor ^ nums[i]);
        dfs(nums, i + 1, curXor);
    }
}